wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -15,26 +15,32 @@
  */
 package com.squareup.wire.gradle
 
+import com.squareup.wire.VERSION
+import com.squareup.wire.gradle.kotlin.sourceRoots
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.plugins.JavaBasePlugin
-import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.artifacts.UnknownConfigurationException
 import org.gradle.api.tasks.TaskProvider
-import org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper
+import org.gradle.api.tasks.compile.JavaCompile
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.plugin.sources.DefaultKotlinSourceSet
 import java.io.File
+import java.util.concurrent.atomic.AtomicBoolean
 
 class WirePlugin : Plugin<Project> {
-  private var kotlin = false
-  private var java = false
+  private val android = AtomicBoolean(false)
+  private val java = AtomicBoolean(false)
+  private val kotlin = AtomicBoolean(false)
```

Comment:
why AtomicBoolean?  I don't typically see this used in Gradle plugins, is this a sign of a race condition we should otherwise avoid?

Reply:
I am not aware of why, I was inspired by SqlDelight here

Reply:
I'm reluctant to copy the SqlDelight implementation here.

How about not merging in the variant code in this PR?

We'd still accomplish 2 of the 3 objectives:
New: Adds wire-runtime automaticallly when there are JVM targets
New: Adds generated code directories into sourceSets automatically